### PR TITLE
Fix terminfo compilation problems so that it compiles on modern machines

### PR DESCRIPTION
**Pull request by lokedhs**

### CODE REVIEW

File: supdup.c [hunk 1]
```diff
@@ -84,12 +85,12 @@ extern char *tgetstr();
...
 
-#define OUTSTRING_BUFSIZ 2048
-unsigned char *outstring;
-
 unsigned char *tparam(), *tgoto();
 #endif /* TERMCAP */
 
+#define OUTSTRING_BUFSIZ 2048
+unsigned char *outstring;
+
```
Comment (larsbrinkhoff) [inline supdup.c:93]: Why move this?

**Comment by larsbrinkhoff** [comment]
I see you removed some of @bictorv's additions.

File: supdup.c (hunk 1 above)
Comment (lokedhs) [reply supdup.c:93]: Because it was used in both the TERMCAP and the TERMINFO branches, but only declared in the TERMCAP branch

**Comment by lokedhs** [comment]
I don't know what those additions were, but there was some hardcoded references to some TERMCAP capabilities. It should be completely safe to fall back to the alternative code path.

**Comment by larsbrinkhoff** [comment]
I have tested that this works in Ubuntu 14.  I haven't looked closely at how the changes affect using TERMCAP.

Since no one has raised any objections, I'll merge this soon.

**Pull request merged**